pandas/core/indexing.py

```diff
@@ -766,9 +766,11 @@ def _getitem_lowerdim(self, tup: Tuple):
         # ...but iloc should handle the tuple as simple integer-location
         # instead of checking it as multiindex representation (GH 13797)
         if isinstance(ax0, ABCMultiIndex) and self.name != "iloc":
-            result = self._handle_lowerdim_multi_index_axis0(tup)
-            if result is not None:
+            try:
+                result = self._handle_lowerdim_multi_index_axis0(tup)
                 return result
```

Comment:
It does, but the problem I tried to address was the default return value of None, as it is checked as a condition later on the _getitem_lowerdim function.

I'm not quite acquainted with the code yet but it seems that for some cases the KeyError won't actually raise so it falls back to the default return of None, which causes the issue here.

Reply:
And I agree, raising just an `Exception` is not very useful, maybe raising an `IndexingError` would be more appropriate?